### ISSUE TITLE
allow to make linked and non-linked duplicates

### ISSUE THREAD

**Issue by mirlip** [s-leger/blenderPolygons#7]
Shift+D and Alt+D could create non-linked and linked duplicates of the parametric objects. Linked one would use the same values (dimensions, angles, etc.) so that modifying one instance would change all the other.
For this, a customisable preset system would maybe be the best. It would also allow to reuse for example a window in several projects.

**Comment by s-leger** [comment]
Yes, 
A preset system need becomes more obvious with stair and the tons of variables you are able to setup.

Archipack objects are build from scratch to support ALT + D and Shift + D. 
Copy to selected is also fully supported.

A refresh button does appair on linked ones to keep missing parts in synch just after duplicate.
Also an unlink button does show on linked objects to make unlinked ones.

Relink objects data to get linked ones does also work.
For multi-sub objects like doors and window, the base object to link is the frame.

**Comment by mirlip** [comment]
Awesome :) I selected the Boolean object to duplicate, that was the reason. This is why I think only one object should be selectable.
It works indeed really well. Some remarks: 
- at the moment, the origin is in the center. It would help to be able to choose the position like bottom/top or right/left (here a standard would have to be defined, but an arbitrary decision is ok). When you modify the length of a window for example, the one you manipulate in UI only grows in the chosen direction, but the duplicates grow from the middle in both direction because the origin is in the middle.
- the addon crashes when I try to go to interactive mode after duplicating the default window on the same wall. Here is the log:
```
Traceback (most recent call last):
  File "C:\Blender\2.78\scripts\addons_extern\blenderPolygons\archipack_autoboolean.py", line 241, in execute
    self.autoboolean(context, wall)
  File "C:\Blender\2.78\scripts\addons_extern\blenderPolygons\archipack_autoboolean.py", line 81, in autoboolean
    bpy.ops.archipack.reference_point()
  File "C:\Blender\2.78\scripts\modules\bpy\ops.py", line 189, in __call__
    ret = op_call(self.idname_py(), None, kw)
RuntimeError: Operator bpy.ops.archipack.reference_point.poll() failed, context is incorrect

location: <unknown location>:-1
```

I had some other bugs, I'll try to make them reproducible and provide the needed files.

**Comment by mirlip** [comment]
Should I close this ticket and open one for the bug and another one about presets ?

**Comment by s-leger** [comment]
Seem i allredy fixed this bug, should investigate about it a bit.

A bug report open for the way linked windows are growing when manipulated from wall.

The wall takes care of windows and doors alignment and depth when autoboolean is used and when wall does change.
I left the pivot intentionnaly at center so you are able to flip direction easly with a r+z+180

**Comment by mirlip** [comment]
Really helpful to have windows and doors aligned automatically to walls. It could allow to have doors and windows to flip with inside/outside of the wall automatically.

**Comment by s-leger** [comment]
It does allready with respect to the side they are facing. 
As long as pivot is near wall (by a factor of 2*wall depth) the tool takes care of alignment and depth.
If you see the demo video, you'll notice i'm simply slapping the door over the wall and that's it.
It's no so obvious as i show now stair works in the between.

**Comment by s-leger** [comment]
Preset system for styles implemented, will be part of 1.2 release
Fixed context bug
